zero_bin/leader/src/client.rs

```diff
@@ -1,19 +1,20 @@
-use std::io::Write;
-use std::path::PathBuf;
 use std::sync::Arc;
 
 use alloy::rpc::types::{BlockId, BlockNumberOrTag, BlockTransactionsKind};
 use alloy::transports::http::reqwest::Url;
-use anyhow::Result;
+use anyhow::{anyhow, Result};
 use paladin::runtime::Runtime;
 use proof_gen::proof_types::GeneratedBlockProof;
-use prover::ProverConfig;
+use prover::{BlockProverInput, ProverConfig};
 use rpc::{retry::build_http_retry_provider, RpcType};
-use tracing::{error, info, warn};
-use zero_bin_common::block_interval::BlockInterval;
-use zero_bin_common::fs::generate_block_proof_file_name;
+use tokio::sync::mpsc;
+use tracing::info;
+use zero_bin_common::block_interval::{BlockInterval, BlockIntervalStream};
 use zero_bin_common::pre_checks::check_previous_proof_and_checkpoint;
 
+// Use some arbitrary number for the channel size, adjust if needed.
+const BLOCK_CHANNEL_SIZE: usize = 16;
```

Comment:
I'd like a slightly more expanded explanation of what is this channel here, and convert it to a TODO for the `adjust if needed` (so that it doesn't get forever forgotten)